packages/@lwc/template-compiler/package.json

```diff
@@ -21,9 +21,9 @@
     "devDependencies": {
         "@types/estree": "0.0.50",
         "@types/he": "^1.1.1",
-        "@types/parse5": "^6.0.1"
+        "@types/parse5": "^6.0.2"
     },
-    "//": "Currently can't upgrade TypeScript to v4.4.2 because it breaks Rollup: https://git.io/JuCcs",
+    "//": "Currently can't upgrade estree-walker to v3.0.0 because it dropped CommonJS support: https://git.io/JXguS",
```

Comment:
This comment was wrong; it was supposed to be about estree-walker, not TypeScript: https://git.io/JXguS

Reply:
Outside the requiring now node@12, do you see any other issue with bumping to `estree-walker`? I would love for us to move start dropping CJS and move to a pure ESM world when possible. 

I was thinking about starting migrating the `observable-membrane` package first to get familiar with this and then update the main `lwc` repo.

Reply:
The main problem with upgrading `estree-walker` to 3.0 is that it breaks the Karma tests. I haven't yet looked into how to fix it.

<details><summary>karma error</summary>

```
$ yarn test
yarn run v1.22.17
$ karma start ./scripts/karma-configs/test/local.js --single-run --browsers Chrome
09 11 2021 08:53:03.675:ERROR [config]: Error in config file!
  Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /Users/nlawson/workspace/lwc/packages/@lwc/template-compiler/node_modules/estree-walker/package.json
    at new NodeError (internal/errors.js:322:7)
    at throwExportsNotFound (internal/modules/esm/resolve.js:322:9)
    at packageExportsResolve (internal/modules/esm/resolve.js:511:7)
    at resolveExports (internal/modules/cjs/loader.js:450:36)
    at Function.Module._findPath (internal/modules/cjs/loader.js:490:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/nlawson/workspace/lwc/packages/@lwc/template-compiler/dist/commonjs/codegen/scope.js:29:25)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>

The other problem is that it would be a breaking change for Node versions that don't support ES modules. Probably not a big deal, but it may be worthwhile checking if all of our big consumers are on [supported Node versions](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-move-my-commonjs-project-to-esm).